### PR TITLE
Adding instructions and scripts to facilitate running E2E tests locally

### PR DESCRIPTION
**Pull request by JimmyCYJ**
Merge minikube-kubernetes-istio/ and vagrant-kubernetes-istio/ into master branch and keep future development work in master branch.

### CODE REVIEW

File: minikube-kubernetes-istio/install_prereqs_debian.sh [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+# Install minikube.
+echo "Checking and Installing Minikube as required"
+minikube --help > /dev/null
+if [ $? -ne 0 ]; then
+  curl -Lo minikube https://storage.googleapis.com/minikube/releases/v0.27.0/minikube-linux-amd64 && chmod +x minikube && sudo mv minikube /usr/local/bin/
+  if [ $? -ne 0 ]; then
+      echo "Looks like minikube installation failed."
+      echo "Please install it manually and then run this script again."
+      exit 1
+    fi
```
Comment (kimikowang) [inline minikube-kubernetes-istio/install_prereqs_debian.sh]: minor: indentation of the scripts is not consistent. Would be good to make them look formatted.

File: vagrant-kubernetes-istio/Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  # Enable provisioning with a shell script. Additional provisioners such as
+  # Puppet, Chef, Ansible, Salt, and Docker are also available. Please see the
+  # documentation for more information about their specific syntax and use.
+  config.vm.provision "shell", inline: <<-SHELL
+     echo "provisioning vagrant vm..."
+     apt-get update
+     echo "install git"
+     sudo apt-get install --quiet -y git
+     echo "install go"
+     sudo curl -O https://dl.google.com/go/go1.10.1.linux-amd64.tar.gz
```
Comment (kimikowang) [inline vagrant-kubernetes-istio/Vagrantfile]: Is it possible to get the link of the latest release version like kubectl?

**Comment by kimikowang** [comment]
@hklai Moving under e2e dir makes sense for me.

File: vagrant-kubernetes-istio/Vagrantfile (hunk 2 above)
Comment (gargnupur) [reply vagrant-kubernetes-istio/Vagrantfile]: Doesn't seem like there is a way here: https://golang.org/dl/

Comment (JimmyCYJ) [reply vagrant-kubernetes-istio/Vagrantfile]: I find a way to download it. Thanks for the advice.

**Comment by JimmyCYJ** [comment]
Please take a look. Thanks!

File: tests/e2e/README.md [hunk 3]
```diff
@@ -197,6 +197,14 @@ delay and version migration. Each test case applies specific rules for itself an
 
 You can build and run this or any single test manually with the same options when testing specific version of master, mixer or istioctl
 
+# Running E2E tests on your local Vagrant VM
```
Comment (hklai) [inline tests/e2e/README.md]: Minor wording suggestion:


# Running E2E tests using a local VM
E2E tests can be run on your local machines as well. It helps local testing and debuging.

See [vagrant-kubernetes-istio/README](vagrant-kubernetes-istio/README.md) for instructions to set up a local Vagrant VM environment to run E2E tests.

And see [minikube-kubernetes-istio/README](minikube-kubernetes-istio/README.md) for instructions to set up a Minikube VM environment to run E2E tests.

File: tests/e2e/minikube-kubernetes-istio/OWNERS [hunk 4]
```diff
@@ -0,0 +1,4 @@
+gargnupur
```
Comment (hklai) [inline tests/e2e/minikube-kubernetes-istio/OWNERS]: I suspect we can use simpler directory name. =)

perhaps something like
test/e2e/local/vagrant
test/e2e/local/minikube

File: tests/e2e/minikube-kubernetes-istio/README.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+sh install_prereqs_debian.sh
+```
+
+# Steps
+## 1. Set up Minikube Environment
+```bash
+sh setup_host.sh
+```
+
+## 2. Build istio images
```
Comment (hklai) [inline tests/e2e/minikube-kubernetes-istio/README.md]: @kimikowang Perhaps you can incorporate your local registry here and simplify the push step?

File: tests/e2e/local/vagrant/revert_dockerdaemon_linux.sh [hunk 6]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
```
Comment (rshriram) [inline tests/e2e/local/vagrant/revert_dockerdaemon_linux.sh]: Why do you need the Mac OS docker daemon png file?

Comment (kimikowang) [reply tests/e2e/local/vagrant/revert_dockerdaemon_linux.sh]: To show users where to add the insecure registry option more clearly.

File: tests/e2e/minikube-kubernetes-istio/OWNERS (hunk 4 above)
Comment (JimmyCYJ) [reply tests/e2e/minikube-kubernetes-istio/OWNERS]: Thanks for the advice, done.

File: tests/e2e/README.md (hunk 3 above)
Comment (JimmyCYJ) [reply tests/e2e/README.md]: Done. Thanks!

**Comment by JimmyCYJ** [comment]
Please take a look @hklai

**Comment by hklai** [comment]
One more q, can the vagrant file in this PR replace
https://github.com/linsheng9731/istio/blob/master/tools/vagrant/Vagrantfile
?

**Pull request merged**